modules/apps/transfer/types/msgs.go

```diff
@@ -69,11 +69,15 @@ func NewMsgTransfer(
 // NOTE: The recipient addresses format is not validated as the format defined by
 // the chain is not known to IBC.
 func (msg MsgTransfer) ValidateBasic() error {
-	if err := host.PortIdentifierValidator(msg.SourcePort); err != nil {
-		return errorsmod.Wrap(err, "invalid source port ID")
-	}
-	if err := host.ChannelIdentifierValidator(msg.SourceChannel); err != nil {
-		return errorsmod.Wrap(err, "invalid source channel ID")
+	// Source ports and channel are overwritten when unwinding, so there's no need
+	// to validate them.
+	if !msg.Forwarding.Unwind {
```

Comment:
It might be good to enforce that if `unwind` is true, then the source port and channel should be empty, because the validation that is done with `PortIdentifierValidator` and `ChannelIdentifierValidator` also checks the length. So this means that somebody could set `unwind` to true and put a ridiculously long string in the source port and channel and then we have again issues with spamming.